### PR TITLE
BufCudaRT allow zero size memory allocation

### PR DESCRIPTION
**Pull request by psychocoderHPC**
- remove asserts those avoid to allocate zero myte memory
- this is a follow up of #185

### CODE REVIEW

File: include/alpaka/mem/buf/BufCudaRt.hpp [hunk 1]
```diff
@@ -436,8 +435,7 @@ namespace alpaka
                                 &pitchBytes,
                                 widthBytes,
                                 height));
-                        assert(memPtr);
-                        assert(pitchBytes>=widthBytes);
+                        assert(pitchBytes>=widthBytes||(width*height)==0);
```
Comment (BenjaminW3) [inline include/alpaka/mem/buf/BufCudaRt.hpp]: Do you know which pitch cudaMallocPitch returns if the width = 0?
If the pitch is also 0, the original assert would be enough and would not need to be extended.

**Comment by BenjaminW3** [comment]
Thanks for finding those!
If we could execute the CUDA enabled test in the CI, those omissions would have been found by the unit tests.

File: include/alpaka/mem/buf/BufCudaRt.hpp (hunk 1 above)
Comment (psychocoderHPC) [inline include/alpaka/mem/buf/BufCudaRt.hpp]: I am not sure. If the with is zero the old assert is enough but what is if `height` is zero. Therefore I added the second check.

**Comment by psychocoderHPC** [comment]
Could we remove the experimentel fiber checks from `appveyor`?

**Comment by BenjaminW3** [comment]
Yes we can do this. As the compilation is currently failing, the error message printing slows down compilation immensly. You can issue a new PR and set [this line](https://github.com/ComputationalRadiationPhysics/alpaka/blob/develop/appveyor.yml#L72) to `OFF`.

**Pull request merged**